integrations/pjrt/cmake/protobuf_cc_library.cmake

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
+include(FetchContent)
+
+# disable some targets we don't use
+set(protobuf_INSTALL OFF)
+set(protobuf_BUILD_TESTS OFF)
+
+# to prevent protobuf itself from using `find_package`
+set(protobuf_FORCE_FETCH_DEPENDENCIES ON)
+
+# pin the version of protobuf
+set(protobuf_VERSION 29.1)
+
+FetchContent_Declare(
+  protobuf
+  GIT_REPOSITORY https://github.com/protocolbuffers/protobuf
+  GIT_TAG v${protobuf_VERSION}
+  GIT_SHALLOW ON
+)
+
+FetchContent_MakeAvailable(protobuf)
+
+# make protobuf_generate() function available
+include(${protobuf_SOURCE_DIR}/cmake/protobuf-generate.cmake)
+
+# protobuf_cc_library()
+#
+# CMake function to invoke the protoc compiler.
+#
+# Parameters:
+# NAME: name of target
+# SRCS: List of source files for the library
+# PROTOC_ARGS: List of protoc arguments.
+# PUBLIC: Add this so that this library will be exported under iree::
+# Also in IDE, target will appear in IREE folder while non PUBLIC will be in IREE/internal.
+# TESTONLY: When added, this target will only be built if user passes -DIREE_BUILD_TESTS=ON to CMake.
+#
+# protobuf_cc_library(
+#   NAME
+#     some_def
+#   SRC
+#     some_def.proto
+#   PUBLIC
+# )
+function(protobuf_cc_library)
+  cmake_parse_arguments(_RULE
+    "PUBLIC;TESTONLY"
+    "NAME"
+    "SRCS;PROTOC_ARGS"
+    ${ARGN}
+  )
+
+  if(_RULE_TESTONLY AND NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  # Prefix the library with the package name, so we get: iree_package_name
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  add_library(${_NAME} ${_RULE_SRCS})
+  protobuf_generate(
+    TARGET ${_NAME}
+    LANGUAGE cpp
+    PROTOC_OPTIONS ${_RULE_PROTOC_ARGS}
+  )
+  target_include_directories(${_NAME}
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+  )
+  target_link_libraries(${_NAME}
+    PUBLIC
+    protobuf::libprotobuf
```

Comment:
(For a separate PR that could land before or after this, not needed in this PR)

Fun fun, this increases the build time from 2 minutes ([logs here](https://github.com/iree-org/iree/actions/runs/12183390132/job/33985228880)) to 8 minutes ([logs here](https://github.com/iree-org/iree/actions/runs/12180379571/job/33975054554?pr=19369)). From 451 build targets to 1035. That's still fast enough to not be too concerned, but we can do better if it starts to impact development iteration time.

We could enable ccache to help a bit there. I'd probably add https://github.com/hendrikmuhs/ccache-action to the workflow, like here: https://github.com/iree-org/iree/blob/d48071df356eec533554699ac1e658f6809aacbe/.github/workflows/ci.yml#L96-L116 and set these environment variables:
```
    export CMAKE_C_COMPILER_LAUNCHER=ccache
    export CMAKE_CXX_COMPILER_LAUNCHER=ccache
```
that might just work...

Reply:
Yeah I also noticed that. It includes about 100 targets to build the protoc compiler for C++ codegen and 400+ targets to build protobuf library and its deps (absl, protobuf-lite..). More than the original total number of jobs. I'll try ccache later : )